### DISCUSSION TITLE
[Feature]: Support IPv6

### DISCUSSION THREAD

**Discussion by pr0ton11** [immich-app/immich#1610] [Feature Request]
### Feature detail

Hi

I am using the Immich Helm Chart in my IPv6 only Kubernetes cluster. Unfortunatly, Immich does not seem to listen to IPv6 at all, making it currently unusable for me.

Is there a parameter where I could tell Immich to bind / listen to an IPv6 address instead?

Thanks you

### Platform

Server

**Comment by samip5** [comment]
Hmm, are you using an Ingress controller on your Kubernetes cluster? If you are, which one and are you still using the proxy container with it (proxy container uses nginx which probably has no IPv6 config)?

**Comment by samip5** [comment]
Just for reference, this is my Immich deployment on my dual-stack cluster with ingress-nginx and no nginx proxy container: <https://github.com/samip5/k8s-cluster/tree/main/cluster/apps/default/immich/app>

**Comment by michelheusschen** [comment]
Probably because the nginx container only listens on IPv4. If you were to change `listen 8080;` with `listen 8080;listen [::]:8080;` in [nginx.conf](https://github.com/immich-app/immich/blob/3cc4af59472321ea5f83fa0679f6a1beb1dd3940/nginx/nginx.conf#L58) it should work after rebuilding the container.

**Comment by samip5** [comment]
It seems that the node process on each container is only listening on IPv4, so that might not be as simple.

It seems that alpine docker container's have IPv6 support disabled:
```
/usr/src/app # sysctl -a 2>/dev/null | grep disable_ipv6
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.eth0.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
```

**Comment by bo0tzz** [comment]
I believe IPv6 should be supported now

**Reply by NMi-ru** [reply]
> I believe IPv6 should be supported now

Latest docker-compose.yml has "ports 2283:2283" which binds to ipv4 only (I run it with lxc, centos 9, podman 5.4.0)

I've added "[::]:2283:2283" to ports; I see that it started to listen on ipv6 as well.

Inside the container, the "immich-api" process listens to ipv6 (can be seen from /proc/net/tcp6), I can initiate a connection from the client (just a simple curl test), but the connection hangs, waiting for response from server:

```
> GET / HTTP/1.1
> Host: test.nmi.ru:2283
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
```
